cmd/juju/caas/add.go

```diff
@@ -152,22 +154,31 @@ func (c *AddCAASCommand) Run(ctxt *cmd.Context) error {
 	if len(caasConfig.Contexts) == 0 {
 		return errors.Errorf("No k8s cluster definitions found in config")
 	}
-	defaultContext := caasConfig.Contexts[caasConfig.CurrentContext]
 
-	defaultCredential := caasConfig.Credentials[defaultContext.CredentialName]
-	defaultCloud := caasConfig.Clouds[defaultContext.CloudName]
+	contextName := c.contextName
+	if contextName == "" {
+		contextName = caasConfig.CurrentContext
+		logger.Debugf("No context name specified, so use current context -> %q", contextName)
+	}
+	context, ok := caasConfig.Contexts[contextName]
+	if !ok {
+		return errors.NotFoundf("contextName -> %q", contextName)
+	}
+
+	credential := caasConfig.Credentials[context.CredentialName]
+	cloud := caasConfig.Clouds[context.CloudName]
```

Comment:
I suspect that the reason you have added an alias to the import is because of the name of this variable.  Coming up with a different variable name feels cleaner than aliasing the import...